src/Features/Core/Portable/ExtractInterface/AbstractExtractInterfaceService.cs

```diff
@@ -147,50 +164,129 @@ public ExtractInterfaceResult ExtractInterfaceFromAnalyzedType(ExtractInterfaceT
+                        extractInterfaceOptions: extractInterfaceOptions,
+                        cancellationToken: cancellationToken).ConfigureAwait(false);
+
+                default: throw new InvalidOperationException($"Unable to extract interface for operation of type {extractInterfaceOptions.GetType()}");
+            }
+        }
 
-            var unformattedInterfaceDocument = GetUnformattedInterfaceDocument(
+        private async Task<ExtractInterfaceResult> ExtractInterfaceToNewFileAsync(
+            Solution solution, string containingNamespaceDisplay, INamedTypeSymbol extractedInterfaceSymbol,
+            ExtractInterfaceTypeAnalysisResult refactoringResult, ExtractInterfaceOptionsResult extractInterfaceOptions, CancellationToken cancellationToken)
+        {
+            var symbolMapping = await CreateSymbolToDeclarationAnnotationMapAsync(
+                extractInterfaceOptions.IncludedMembers,
                 solution,
+                refactoringResult.TypeNode,
+                cancellationToken).ConfigureAwait(false);
+
+
+
+            var interfaceDocumentId = DocumentId.CreateNewId(refactoringResult.DocumentToExtractFrom.Project.Id, debugName: extractInterfaceOptions.FileName);
+
+            var unformattedInterfaceDocument = await GetUnformattedInterfaceDocumentAsync(
+                symbolMapping.AnnotatedSolution,
                 containingNamespaceDisplay,
                 extractInterfaceOptions.FileName,
                 refactoringResult.DocumentToExtractFrom.Folders,
                 extractedInterfaceSymbol,
                 interfaceDocumentId,
-                cancellationToken);
+                cancellationToken).ConfigureAwait(false);
 
-            var solutionWithFormattedInterfaceDocument = GetSolutionWithFormattedInterfaceDocument(unformattedInterfaceDocument, cancellationToken);
+            var solutionWithFormattedInterfaceDocument = await GetFormattedSolutionAsync(
+                unformattedInterfaceDocument.Project.Solution,
+                SpecializedCollections.SingletonList(unformattedInterfaceDocument.Id),
+                cancellationToken).ConfigureAwait(false);
 
-            var completedSolution = GetSolutionWithOriginalTypeUpdated(
+            var completedSolution = await GetSolutionWithOriginalTypeUpdatedAsync(
                 solutionWithFormattedInterfaceDocument,
-                documentIds,
+                symbolMapping.DocumentIds,
                 refactoringResult.DocumentToExtractFrom.Id,
-                typeNodeSyntaxAnnotation,
+                symbolMapping.TypeNodeAnnotation,
                 refactoringResult.TypeToExtractFrom,
                 extractedInterfaceSymbol,
                 extractInterfaceOptions.IncludedMembers,
-                symbolToDeclarationAnnotationMap,
-                cancellationToken);
+                symbolMapping.SymbolToDeclarationAnnotationMap,
+                cancellationToken).ConfigureAwait(false);
```

Comment:
can this helper just take the symbolMapping instance?

Reply:
Diatribe is in the other comment, but I prefer to keep `SymbolMapping` a private type, and abstract implementer shouldn't know or care about it.